eqcorrscan/scripts/correlation_speeds.py

```diff
@@ -0,0 +1,176 @@
+"""
+This script will construct a synthetic dataset of specified size and run the
+various correlation functions available in EQcorrscan over the data for
+parallelisms compatible with your machine.  In doing so timings and memory
+usage will be computed to provide information to the user as to what the best
+algorithm for their dataset and architecture is.
+
+:copyright:
+    EQcorrscan developers.
+
+:license:
+    GNU Lesser General Public License, Version 3
+    (https://www.gnu.org/copyleft/lesser.html)
+"""
+
+import time
+import resource
+import sys
+import numpy as np
+import matplotlib.pyplot as plt
+import itertools
+
+from memory_profiler import memory_usage
+from obspy import Trace, Stream
+from multiprocessing import cpu_count
+
+from eqcorrscan.utils.correlate import (
+    XCOR_FUNCS, XCORR_STREAM_METHODS, get_stream_xcorr)
+from eqcorrscan.utils import correlate
+
+
+def memory_limit():
+    soft, hard = resource.getrlimit(resource.RLIMIT_AS)
+    resource.setrlimit(resource.RLIMIT_AS, (get_memory() * 1024, hard))
+
+
+def get_memory():
+    with open('/proc/meminfo', 'r') as mem:
```

Comment:
Yup, this definitely isn't - psutil will be my friend here, but the function above that uses the `resource` module is more of a pain - `resource` is a standard Python library, but it is Unix only :sob:. I will have to think of another way to limit memory usage on windows.  I may end up writing a wrapper that queries `psutil.virtual_memory` every n seconds, but that will slow things down a bit.  That shouldn't be too much of a issue because that slow-sown should be the same for every function so the times will be systematically shifted, which should not preclude choosing the fastest.